### PR TITLE
LibJS: Use SortIndexedProperties AO in [Typed]Array.prototype.sort

### PR DESCRIPTION
**Pull request by trflynn89**
This started with looking to see if we needed to do anything with https://github.com/tc39/ecma262/commit/70baf25 (we don't), and noticed we can be closer to the spec with these methods.

### CODE REVIEW

File: Userland/Libraries/LibJS/Runtime/ArrayPrototype.cpp [hunk 1]
```diff
@@ -1551,42 +1551,62 @@ ThrowCompletionOr<void> array_merge_sort(GlobalObject& global_object, FunctionOb
...
-        TRY(object->set(j, items[j], Object::ShouldThrowExceptions::Yes));
+    // 9. Repeat, while j < itemCount,
+    for (; j < item_count; ++j) {
+        // a. Perform ? CreateDataPropertyOrThrow(obj, ! ToString(𝔽(j)), sortedList[j]).
 
-    // The empty parts of the array are always sorted to the end, regardless of the
-    // compare function. FIXME: For performance, a similar process could be used
-    // for undefined, which are sorted to right before the empty values.
-    for (size_t j = items.size(); j < length; ++j)
+        // FIXME: Spec issue: The above step should likely be:
```
Comment (linusg) [inline Userland/Libraries/LibJS/Runtime/ArrayPrototype.cpp:1587]: Are you filing an issue for this? :)

**Review by linusg — APPROVED** [review]
:poggie:

File: Userland/Libraries/LibJS/Runtime/ArrayPrototype.cpp (hunk 1 above)
Comment (trflynn89) [reply Userland/Libraries/LibJS/Runtime/ArrayPrototype.cpp:1587]: https://github.com/tc39/proposal-change-array-by-copy/issues/97

Comment (trflynn89) [reply Userland/Libraries/LibJS/Runtime/ArrayPrototype.cpp:1587]: I'll link that in the comment here for reference too

Comment (linusg) [reply Userland/Libraries/LibJS/Runtime/ArrayPrototype.cpp:1587]: Nice, ty!

**Pull request merged**